### PR TITLE
Feat google geofence

### PR DESCRIPTION
**Pull request by reidhurlburt**
Add all the geofence functionality.

I've noticed that a notification saying the app is not responding pops up every now and then. If that starts happening to everyone and not just my 8GB DDR3 laptop, then it's probably in here.

### CODE REVIEW

**Comment by reidhurlburt** [comment]
> Add all the geofence functionality.
> 
> I've noticed that a notification saying the app is not responding pops up every now and then. If that starts happening to everyone and not just my 8GB DDR3 laptop, then it's probably in here.

Looks like it was just the state of my emulator, disregard

File: lib/gateway.dart [hunk 1]
```diff
@@ -270,6 +274,106 @@ class Gateway {
     return null;
   }
 
+  Future<Object?> addParkingLot(List<GeoCoord> coords, String name) async {
+    final ParkingLot lot = ParkingLot(name: name, coords: coords);
+    GraphQLRequest<Organization> request;
+    GraphQLResponse<Organization> response;
+    try {
+      Organization? organization = await getOrganization('unb');
```
Comment (ChrisRuff) [inline lib/gateway.dart:282]: We dont hardcode UNB, use CurrentUser().getOrg()

File: lib/gateway.dart [hunk 2]
```diff
@@ -270,6 +274,106 @@ class Gateway {
...
+        print('Mutation failed: $e');
+      }
+    }
+  }
+
+  Future<Object?> removeParkingLots() async {
+    GraphQLRequest<Organization> request;
+    GraphQLResponse<Organization> response;
+    try {
+      Organization? organization = await getOrganization('unb');
```
Comment (ChrisRuff) [inline lib/gateway.dart:306]: See comment above

File: lib/gateway.dart [hunk 3]
```diff
@@ -270,6 +274,106 @@ class Gateway {
...
+      polygon.add(LatLng(coords?.elementAt(i).latitude ?? 0.0, coords?.elementAt(i).longitude ?? 0.0));
+    }
+    return polygon;
+  }
+
+  Future<Object?> inParkingLot(LocationData? curLocation, String org) async {
+    GraphQLRequest<Organization> request;
+    GraphQLResponse<Organization> response;
+    try {
+      Organization? organization = await getOrganization('unb');
```
Comment (ChrisRuff) [inline lib/gateway.dart:359]: See comment above x2

File: lib/views/homepage.dart [hunk 4]
```diff
@@ -170,11 +173,19 @@ class _HomePageState extends State<HomePage> {
...
                               null,
                               null) as Registration?;
-                          // STUB
-                          // examine(plate_ctrl.text);
+
+                          var curLoc = await location.getLocation();
+
+                          final loc = await loadingDialog(
+                              context,
+                              Gateway().inParkingLot(curLoc, 'unb'),
```
Comment (ChrisRuff) [inline lib/views/homepage.dart:181]: We dont need to hardcode org, use CurrentUser().getOrg()

File: lib/gateway.dart [hunk 5]
```diff
@@ -270,6 +274,106 @@ class Gateway {
...
+      if (organization?.parkingLots != null) {
+        for(int i=0; i<(organization?.parkingLots?.length ?? 0); i++) {
+          if (PolygonUtil.containsLocation(LatLng(curLocation?.latitude ?? 0.0,
+              curLocation?.longitude ?? 0.0), convertGeoCoordLatLng(
+              organization?.parkingLots?.elementAt(i).coords), false)) {
+            return organization?.parkingLots?.elementAt(i)?.name;
+          }
```
Comment (ChrisRuff) [inline lib/gateway.dart:366]: The dart null dance above is ridiculous, but I approve

File: lib/main.dart [hunk 6]
```diff
@@ -83,6 +89,33 @@ Future<void> setupAmplify() async {
...
+bool _serviceEnabled = false;
+PermissionStatus? _permissionGranted;
+late LocationData locationData;
+
+_checkLocationPermission() async {
+  _serviceEnabled = await location.serviceEnabled();
+  if (!_serviceEnabled) {
+    _serviceEnabled = await location.requestService();
+    if (!_serviceEnabled) {
+      print("Service is f'd");
```
Comment (ChrisRuff) [inline lib/main.dart:102]: Lets clean up some prints

File: lib/main.dart [hunk 7]
```diff
@@ -83,6 +89,33 @@ Future<void> setupAmplify() async {
...
+      return;
+    }
+  }
+  _permissionGranted = await location.hasPermission();
+  if (_permissionGranted == PermissionStatus.denied) {
+    _permissionGranted = await location.requestPermission();
+    if (_permissionGranted != PermissionStatus.granted) {
+      print("Permission is f'd");
```
Comment (ChrisRuff) [inline lib/main.dart:110]: Clean up prints

**Review by ChrisRuff — CHANGES_REQUESTED** [review]

**Review by ChrisRuff — APPROVED** [review]

**Pull request merged**